### PR TITLE
[shopsys] Improve upgrade instructions 

### PR DESCRIPTION
**Pull request by grossmannmartin**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Improve upgrade notes resulting in more understandable instructions, making it easier for implementers to upgrade. Define clear rules for writing upgrade instructions.
|New feature| No 
|BC breaks| May break releaser tool 
|Fixes issues| ... 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: docs/contributing/guidelines-for-writing-upgrade.md [hunk 1]
```diff
@@ -17,3 +19,54 @@ Keep in mind that upgrade instructions are written for users that do not underst
 * Write instructions
     * Good example: *"Do this, then that"*
     * Bad example: *"This was done, this was changed"*
+
+## Files related to upgrade
+
+The main file to start looking for instructions must be [`UPGRADE.md`](../../UPGRADE.md).
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply docs/contributing/guidelines-for-writing-upgrade.md]: The sentence sounds better, thanks 👍 

Absolute paths are indeed more readable, but then PhpStorm can not suggest files. And even GitHub recommend relative links https://help.github.com/articles/basic-writing-and-formatting-syntax/#relative-links. 
But I'm open to discussion :)

File: docs/upgrade/UPGRADE-7.0.0-alpha2.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    - easiest way is to overwrite your `docker-compose.yml` with by the appropriate template
+- on *nix systems, fill your UID and GID (you can run `id -u` and `id -g` to obtain them) into Docker build arguments `www_data_uid` and `www_data_gid` and rebuild your image via `docker-compose up --build`
+- change owner of the files in shared volume to `www-data` from the container by running `docker exec -u root shopsys-framework-php-fpm chown -R www-data /var/www/html`
+    - the user has shared UID, so you will be able to access it as well from the host machine
+    - shared volume with postgres data should be owned by `postgres` user: `docker exec -u root shopsys-framework-php-fpm chown -R postgres /var/www/html/var/postgres-data`
+- if you were using a mounted volume to share Composer cache with the container, change the target directory from `/root/.composer` to `/home/www-data/.composer`
+    - in such case, you should change the owner as well by running `docker exec -u root shopsys-framework-php-fpm chown -R www-data /home/www-data/.composer`
+
+[shopsys/shopsys]: https://github.com/shopsys/shopsys
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply docs/upgrade/UPGRADE-7.0.0-alpha2.md]: You're right

File: docs/upgrade/UPGRADE-7.0.0-alpha2.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#UPGRADE
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply docs/upgrade/UPGRADE-7.0.0-alpha2.md]: Oh, I completely missed that. Okay, I've changed headings to format "Upgrade from v7.0.0-beta3 to v7.0.0-beta4" and made a 1st level heading of them. 
I also changed level of other headings appropriately.

File: docs/upgrade/UPGRADE-7.0.0-alpha2.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply docs/upgrade/UPGRADE-7.0.0-alpha2.md]: What about shopsys/framework link? 
Should it point to framework, project-base or monorepo? Framework doesn't feel enough as there are instructions for project-base too.

File: docs/upgrade/UPGRADE-7.0.0-beta3.md [hunk 4]
```diff
@@ -1,10 +1,177 @@
 ## [From 7.0.0-beta2 to v7.0.0-beta3](https://github.com/shopsys/shopsys/compare/v7.0.0-beta2...v7.0.0-beta3)
 
+### [shopsys/coding-standards]
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply docs/upgrade/UPGRADE-7.0.0-beta3.md]: We should keep ordering from UPGRADE.md. But the question is, should be framework first? 
Coding-standards makes sense to me to be first. But I don't know about others, though. 
You think FW should be always first?

File: docs/upgrade/UPGRADE-7.0.0-beta4.md [hunk 5]
```diff
@@ -1,23 +1,12 @@
...
 
-[shopsys/shopsys]: https://github.com/shopsys/shopsys
-[shopsys/project-base]: https://github.com/shopsys/project-base
+### [shopsys/framework]
+
+#### Infrastructure
+- set `ENV COMPOSER_MEMORY_LIMIT=-1` in base stage in your `docker/php-fpm/Dockerfile` as composer consumes huge amount of memory during dependencies installation ([#635](https://github.com/shopsys/shopsys/pull/635/files))
+
+#### Configuration
+- modify your `src/Shopsys/ShopBundle/Resources/config/services.yml` ([#616](https://github.com/shopsys/shopsys/pull/616))
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply docs/upgrade/UPGRADE-7.0.0-beta4.md]: It wasn't marked *(optional)* by the author, so I left it like it was. It's definitely an optional step, but we can count on this in some workshop or cookbook. But "maybe" is not the proper reason. 

I make it optional.

File: docs/contributing/UPGRADE.md [hunk 6]
```diff
@@ -1,2 +1,92 @@
-# UPGRADING A MONOREPO
+# UPGRADING MONOREPO
 
+## [From v7.0.0-beta4 to Unreleased](https://github.com/shopsys/shopsys/compare/v7.0.0-beta4...HEAD)
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply docs/contributing/UPGRADE.md]: You're right once again 👍 I'll be more careful next time.

File: docs/contributing/UPGRADE.md [hunk 7]
```diff
@@ -1,2 +1,92 @@
...
+    - the Dockerfile for `php-fpm` has changed, update your `docker-compose.yml` and `docker/php-fpm/Dockerfile` accordingly
+        - copy [`docker/php-fpm/Dockerfile`](https://github.com/shopsys/shopsys/blob/master/project-base/docker/php-fpm/Dockerfile) from GitHub
+        - update the build config in `docker-compose.yml` ([changes in version and build config can be seen in the PR](https://github.com/shopsys/shopsys/pull/533/files#diff-1aa104f9fc120d0743883a5ba02bfe21))
+    - rebuild images by running `docker-compose up -d --build`
+- *(optional)* rename Database tests to Functional tests
+    - rename base class `DatabaseTestCase` to `TransactionFunctionalTestCase`
+    - rename test namespace `Database` to `Functional`
+    - rename phing target `tests-db` to `tests-functional`
+    - you can follow [#541 Rename database tests to functional tests](https://github.com/shopsys/shopsys/pull/541)
+- [#530 - Update of installation for production via docker](https://github.com/shopsys/shopsys/pull/530)
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply docs/contributing/UPGRADE.md]: I truly just moved shopsys/shopsys section, but it will be nice to clean here too 👍

File: docs/contributing/guidelines-for-writing-upgrade.md [hunk 8]
```diff
@@ -17,3 +19,58 @@ Keep in mind that upgrade instructions are written for users that do not underst
...
     * Good example: *"Do this, then that"*
     * Bad example: *"This was done, this was changed"*
+* If there are any upgrading steps that are not vital for project developers but could be helpful for them, 
+    they should be mentioned in the upgrading notes as well and marked as (optional). 
+    The general UPGRADE.md then mentions that the optional steps are not vital, 
+    however we recommend to perform them as well during upgrading as it might ease their work in the future
+
+## Files related to upgrade
+
+The main file where a project developer should start looking for is [`UPGRADE.md`](../../UPGRADE.md) file in the monorepo root.
```
Comment (vitek-rostislav) [inline docs/contributing/guidelines-for-writing-upgrade.md]:
```suggestion
The main file where a project developer should start looking for instructions is [`UPGRADE.md`](../../UPGRADE.md) file in the monorepo root.
```

File: docs/contributing/guidelines-for-writing-upgrade.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (vitek-rostislav) [reply docs/contributing/guidelines-for-writing-upgrade.md]: Ok, let us keep the relative paths :slightly_smiling_face: I have added yet one more suggestion as I am missing the word "instructions" now :wink:

File: docs/upgrade/UPGRADE-7.0.0-alpha2.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (vitek-rostislav) [reply docs/upgrade/UPGRADE-7.0.0-alpha2.md]: hm, to me, link to `shopsys/framework` is sufficient. I would link to the other repos only if they were mentioned in the text. The heading says "shopsys/framework", that is why it needs the link. It is true that there are instructions for `project-base` as well, however, these instructions are steps that you (as a developer of a project based on project-base) should do in your project when upgrading the dependency on `shopsys/framework` :slightly_smiling_face: So I do not feel, that the `project-base` repo needs to be linked in such a case.

File: docs/upgrade/UPGRADE-7.0.0-alpha2.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (vitek-rostislav) [reply docs/upgrade/UPGRADE-7.0.0-alpha2.md]: I do not want to be a nitpick, however, you missed `before-monorepo.md` yet, it begins with 2nd level heading :slightly_smiling_face:

File: docs/upgrade/UPGRADE-7.0.0-beta3.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (vitek-rostislav) [reply docs/upgrade/UPGRADE-7.0.0-beta3.md]: Well, yes, the framework should be always first from my point of view - it is probably the most important package that our users will be upgrading (or, in other words, the most common use-case). 

However, as it is quite hard to maintain, I would be perfectly fine with removing the ordered list from `UPGRADE.md` completely a do not care about the order at all. 

Now, when the instructions are divided into separate files for each version, it is much clearer and I do not see the necessity for that ordering. 

Moreover, the sentence does not make sense anymore as `shopsys/shopsys` and `shopsys/project-base` are not mentioned in `UPGRADE-xx` files anymore but we can not remove them from the list as it states "There is a list of **all** the repositories maintained by monorepo..."

So, all in all, if you agree with me, I suggest you removing the list along with the sentence from `UPGRADE.md` and do not care about the order of packages anymore.

However, beware, if I remember correctly, @LukasHeinz wanted that ordered list there so please discuss it with him during the business validation :slightly_smiling_face:

File: docs/upgrade/UPGRADE-7.0.0-alpha5.md [hunk 9]
```diff
@@ -1,4 +1,4 @@
-## [From 7.0.0-alpha4 to 7.0.0-alpha5](https://github.com/shopsys/shopsys/compare/v7.0.0-alpha4...v7.0.0-alpha5)
+## [Upgrade from 7.0.0-alpha4 to 7.0.0-alpha5](https://github.com/shopsys/shopsys/compare/v7.0.0-alpha4...v7.0.0-alpha5)
```
Comment (vitek-rostislav) [inline docs/upgrade/UPGRADE-7.0.0-alpha5.md]: the level of heading is not fixed

File: docs/upgrade/UPGRADE-7.0.0-alpha6.md [hunk 10]
```diff
@@ -1,4 +1,5 @@
-## [From 7.0.0-alpha5 to 7.0.0-alpha6](https://github.com/shopsys/shopsys/compare/v7.0.0-alpha5...v7.0.0-alpha6)
+## [Upgrade from 7.0.0-alpha5 to 7.0.0-alpha6](https://github.com/shopsys/shopsys/compare/v7.0.0-alpha5...v7.0.0-alpha6)
```
Comment (vitek-rostislav) [inline docs/upgrade/UPGRADE-7.0.0-alpha6.md]: the level of heading is not fixed

File: docs/upgrade/UPGRADE-7.0.0-beta2.md [hunk 11]
```diff
@@ -1,5 +1,6 @@
-## [From 7.0.0-beta1 to 7.0.0-beta2](https://github.com/shopsys/shopsys/compare/v7.0.0-beta1...v7.0.0-beta2)
-### [shopsys/project-base]
+# [From 7.0.0-beta1 to 7.0.0-beta2](https://github.com/shopsys/shopsys/compare/v7.0.0-beta1...v7.0.0-beta2)
```
Comment (vitek-rostislav) [inline docs/upgrade/UPGRADE-7.0.0-beta2.md]: Now I really feel like a nitpick, however, you missed the word "Upgrade" in the beginning here

File: docs/contributing/UPGRADE.md [hunk 12]
```diff
@@ -26,7 +26,7 @@
             src: './project-base/var/elasticsearch-data/'
             host_disk_mount_mode: 'cached'
         ```
-    - *(monorepo only)* add `shopsys-framework-microservice-product-search-sync` and `shopsys-framework-microservice-product-search-export-sync` volumes to `docker-compose.yml` for `php-fpm` service
```
Comment (vitek-rostislav) [inline docs/contributing/UPGRADE.md]: this whole commit should be a fixup! of d6b727a7c :slightly_smiling_face:

File: docs/contributing/guidelines-for-writing-upgrade.md [hunk 13]
```diff
@@ -19,6 +19,10 @@ Keep in mind that upgrade instructions are written for users that do not underst
 * Write instructions
     * Good example: *"Do this, then that"*
     * Bad example: *"This was done, this was changed"*
+* If there are any upgrading steps that are not vital for project developers but could be helpful for them, 
+    they should be mentioned in the upgrading notes as well and marked as (optional). 
+    The general UPGRADE.md then mentions that the optional steps are not vital,
```
Comment (vitek-rostislav) [inline docs/contributing/guidelines-for-writing-upgrade.md]: This is not true yet, you need to add the information there :slightly_smiling_face:

File: docs/contributing/UPGRADE.md [hunk 14]
```diff
@@ -0,0 +1,65 @@
+# UPGRADING MONOREPO
```
Comment (vitek-rostislav) [inline docs/contributing/UPGRADE.md]:
```suggestion
# Upgrading monorepo
```

**Comment by vitek-rostislav** [comment]
Oh, one more thing, please check [Jenkins](http://heimdall:8080/job/mg-upgrade-phase2/6/console) - you have redundant trailing whitespace in `guidelines-for-writing-upgrade.md`

File: docs/upgrade/UPGRADE-7.0.0-beta2.md (hunk 11 above)
Comment (grossmannmartin) [reply docs/upgrade/UPGRADE-7.0.0-beta2.md]: No, it's perfectly ok :) I have to be more careful.

File: docs/upgrade/UPGRADE-7.0.0-beta3.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply docs/upgrade/UPGRADE-7.0.0-beta3.md]: Ok, thank you :)

File: docs/contributing/guidelines-for-writing-upgrade.md (hunk 13 above)
Comment (grossmannmartin) [reply docs/contributing/guidelines-for-writing-upgrade.md]: Yes, atomic commit again. I didn't commit yet. Fixed.

**Comment by grossmannmartin** [comment]
Thank you, @vitek-rostislav for a great review! 
There should be all fixes applied, I hope I didn't forget something. Can you please check the changes?
After that, I rebase and make history nice again.

File: docs/upgrade/UPGRADE-7.0.0-alpha2.md [hunk 15]
```diff
@@ -1,5 +1,9 @@
 # [Upgrade from 7.0.0-alpha1 to 7.0.0-alpha2](https://github.com/shopsys/shopsys/compare/v7.0.0-alpha1...v7.0.0-alpha2)
 
+This guide contain instructions to upgrade from version 7.0.0-alpha1 to 7.0.0-alpha2.
```
Comment (vitek-rostislav) [inline docs/upgrade/UPGRADE-7.0.0-alpha2.md]: :ok_hand:

File: docs/upgrade/UPGRADE-7.0.0-alpha2.md [hunk 16]
```diff
@@ -1,5 +1,9 @@
...
+Before you start, don't forget to take a look at [general instructions](/UPGRADE.md) about upgrading.
```
Comment (vitek-rostislav) [inline docs/upgrade/UPGRADE-7.0.0-alpha2.md]: :ok_hand:

File: UPGRADE.md [hunk 17]
```diff
@@ -10,7 +10,10 @@ Follow instructions in a [monorepo upgrade guide](./docs/contributing/UPGRADE.md
 
 ### You are developing a project based on project-base
 * upgrade only your composer dependencies and follow instructions
-* if you want update your project with the changes from [shopsys/project-base], you can follow the *(optional)* instructions or cherry-pick from the repository whatever is relevant for you but we do not recommend rebasing or merging everything because the changes might not be compatible with your project as it probably evolves in time
+* if you want update your project with the changes from [shopsys/project-base],
+    you can follow the *(optional)* instructions or cherry-pick from the repository whatever is relevant for you but we do not recommend rebasing or merging everything because the changes might not be compatible with your project as it probably evolves in time
+* instructions marked as *(optional)* are not vital, but could be helpful,
```
Comment (vitek-rostislav) [inline UPGRADE.md]: this should be in a separate commit :slightly_smiling_face: it does not fit to "upgrade documentation: Removed ordered list of repositories maintained by monorepo"

Comment (vitek-rostislav) [inline UPGRADE.md]: this should be in a separate commit :slightly_smiling_face: it does not fit to "upgrade documentation: Removed ordered list of repositories maintained by monorepo"

File: UPGRADE.md [hunk 18]
```diff
@@ -10,7 +10,10 @@ Follow instructions in a [monorepo upgrade guide](./docs/contributing/UPGRADE.md
 
 ### You are developing a project based on project-base
 * upgrade only your composer dependencies and follow instructions
-* if you want update your project with the changes from [shopsys/project-base], you can follow the *(optional)* instructions or cherry-pick from the repository whatever is relevant for you but we do not recommend rebasing or merging everything because the changes might not be compatible with your project as it probably evolves in time
+* if you want update your project with the changes from [shopsys/project-base],
```
Comment (vitek-rostislav) [inline UPGRADE.md]: please, do not do such changes if not necessary - the diff is then slightly confusing :slightly_smiling_face: 

The longer lines might be hard to read in PHP Storm when you have preview and navigation windows open as well - for me, [soft wrapping](https://stackoverflow.com/questions/10351608/word-wrapping-in-phpstorm) is helpful from time to time :wink:

File: docs/contributing/UPGRADE.md [hunk 19]
```diff
@@ -0,0 +1,73 @@
+# Upgrading monorepo
```
Comment (LukasHeinz) [inline docs/contributing/UPGRADE.md]: Please reconsider changing name of this file to something like "monorepo upgrade"

File: docs/upgrade/UPGRADE-unreleased.md [hunk 20]
```diff
@@ -0,0 +1,58 @@
+# [Upgrade from v7.0.0-beta4 to Unreleased](https://github.com/shopsys/shopsys/compare/v7.0.0-beta4...HEAD)
+
+This guide contain instructions to upgrade from version 7.0.0-beta4 to Unreleased.
+Before you start, don't forget to take a look at [general instructions](/UPGRADE.md) about upgrading.
+There you can find links to upgrade notes for other versions too.
+
+## [shopsys/migrations]
```
Comment (LukasHeinz) [inline docs/upgrade/UPGRADE-unreleased.md]: shouldn't this section be placed below /framework?

File: docs/contributing/guidelines-for-writing-upgrade.md [hunk 21]
```diff
@@ -2,6 +2,8 @@
 
 Keep in mind that upgrade instructions are written for users that do not understand our system well, so more clear they are, more useful they are.
 
+## Introduction
```
Comment (LukasHeinz) [inline docs/contributing/guidelines-for-writing-upgrade.md]: any rules about linking to PR and that every buletpoint is one PR (or should be)?

File: docs/contributing/guidelines-for-writing-upgrade.md [hunk 22]
```diff
@@ -2,6 +2,8 @@
...
+
```
Comment (LukasHeinz) [inline docs/contributing/guidelines-for-writing-upgrade.md]: I like instructions with with link to specific changes - https://github.com/shopsys/shopsys/blame/d315cfb96b9e9ed344e2b81c9f1deec7c89873b7/docs/upgrade/UPGRADE-unreleased.md#L13

File: docs/upgrade/UPGRADE-7.0.0-beta3.md [hunk 23]
```diff
@@ -0,0 +1,259 @@
...
+- there are a few new standards, i.e. [new fixers enabled](https://github.com/shopsys/shopsys/pull/573/files#diff-709e8469a9fc8c8b45f8b285ac1a4c92) in the `easy-coding-standard.yml` config that enforce using annotations for all your methods:
+    - if you want to use the standards as well, let the fixers check and fix your code
+        - on projects built from `shopsys/project-base` just run `php phing standards-fix`
+        - on any other projects, where you do not have our phing targets available, run `vendor/bin/ecs check path/to/your/source-codes --fix`
+    - if you do not want to use these standards, disable them in your custom configuration (`easy-coding-standard.yml` by default)
+
+## [shopsys/framework]
+### Infrastructure
+- run `docker-compose down` to turn off your containers
+- *(MacOS, Windows only)* run `docker-sync stop`
```
Comment (LukasHeinz) [inline docs/upgrade/UPGRADE-7.0.0-beta3.md]: Is this necessary? It is part of general upgrade instructions,...

File: docs/upgrade/UPGRADE-7.0.0-beta3.md [hunk 24]
```diff
@@ -0,0 +1,259 @@
...
+        ```
+    - replace all occurrences of `/content/` with new parameter `/%shopsys.content_dir_name%/` in `routing-front.yml` and `paths.yml` files
+
+### Database migrations
+- after running database migrations, all your products will be using manual pricing and will have set prices for all pricing groups in a fashion that will keep the final price as same as before
+    - we strongly recommend to review [`Version20181114134959`](https://github.com/shopsys/shopsys/blob/v7.0.0-beta3/packages/framework/src/Migrations/Version20181114134959.php)
+    and [`Version20181114145250`](https://github.com/shopsys/shopsys/blob/v7.0.0-beta3/packages/framework/src/Migrations/Version20181114145250.php) migrations before executing them on your real data,
+    especially if there were any modifications in your product pricing implementation on the project. ([#595](https://github.com/shopsys/shopsys/pull/595))
+
+If any of the migrations does not suit you, there is an option to skip it, see [our Database Migrations docs](https://github.com/shopsys/shopsys/blob/master/docs/introduction/database-migrations.md#reordering-and-skipping-migrations)
```
Comment (LukasHeinz) [inline docs/upgrade/UPGRADE-7.0.0-beta3.md]: This line will be there every time?

File: docs/upgrade/UPGRADE-7.0.0-beta3.md [hunk 25]
```diff
@@ -0,0 +1,259 @@
+# [Upgrade from 7.0.0-beta2 to v7.0.0-beta3](https://github.com/shopsys/shopsys/compare/v7.0.0-beta2...v7.0.0-beta3)
+
+This guide contain instructions to upgrade from version 7.0.0-beta2 to 7.0.0-beta3.
+Before you start, don't forget to take a look at [general instructions](/UPGRADE.md) about upgrading.
+There you can find links to upgrade notes for other versions too.
+
+## [shopsys/coding-standards]
```
Comment (LukasHeinz) [inline docs/upgrade/UPGRADE-7.0.0-beta3.md]: shopsys/framewrk should be first, shouldn't it?

File: docs/upgrade/UPGRADE-7.0.0-beta3.md [hunk 26]
```diff
@@ -0,0 +1,259 @@
...
+    - the last parameter is no longer `bool`, but `integer` - domain ID
+- `Shopsys\FrameworkBundle\Model\Product\ProductFacade::create()` and `Shopsys\FrameworkBundle\Model\Product\ProductFactory` were modified
+    - if you extended the classes in your project, please check out the changes in the framework ones (and the reasons for the changes) in [#581](https://github.com/shopsys/shopsys/pull/581/files)
+- register templates for new FormTypes (DisplayOnlyCustomerType and OrderItemsType) by adding following lines to `app/config/packages/twig.yml` ([#576](https://github.com/shopsys/shopsys/pull/576))
+    ```yaml
+    - '@ShopsysFramework/Admin/Form/orderItems.html.twig'
+    - '@ShopsysFramework/Admin/Form/displayOnlyCustomer.html.twig'
+    ```
+- OrderFormType in administration is now rendered by default ([#576](https://github.com/shopsys/shopsys/pull/576))
+    - to add new field or group is not now necessary to extend twig template, but is sufficient to create OrderFormTypeExtension (see [example on demoshop](https://github.com/shopsys/demoshop/pull/27/commits/c6a54c6592ebab8cba6d86e47985fb31f511ba30#diff-0c583c8b886a9844ffd793f1afa83ec9))
```
Comment (LukasHeinz) [inline docs/upgrade/UPGRADE-7.0.0-beta3.md]: This is only description of the PR, isn't it? Should it be here?

File: UPGRADE.md [hunk 27]
```diff
@@ -6,11 +6,14 @@ Since these are 3 possible scenarios how you can use shopsys, instructions are d
 Follow instructions in relevant sections, eg. `shopsys/coding-standards`, `shopsys/microservice-product-search`.
 
 ### You are using monorepo
-Follow instructions in the section `shopsys/shopsys`.
+Follow instructions in a [monorepo upgrade guide](./docs/contributing/UPGRADE.md)
 
 ### You are developing a project based on project-base
 * upgrade only your composer dependencies and follow instructions
-* if you want update your project with the changes from [shopsys/project-base], you can follow the *(optional)* instructions or cherry-pick from the repository whatever is relevant for you but we do not recommend rebasing or merging everything because the changes might not be compatible with your project as it probably evolves in time
+* if you want update your project with the changes from [shopsys/project-base],
```
Comment (simara-svatopluk) [inline UPGRADE.md]: after reading the whole documentation, I didn't know what instructions are relevant for upgrading project-base. I would recommend to move the bullet `check all instructions in all sections, any of them could be relevant for you` to second place (after `upgrade only your composer dep...`), so it has "higher priority" for reader

File: docs/contributing/guidelines-for-writing-upgrade.md (hunk 21 above)
Comment (grossmannmartin) [reply docs/contributing/guidelines-for-writing-upgrade.md]: There is an instruction in `guidelines-for-writing-upgrade.md` that states 

> Each step should have a link to the related pull request and may contain an additional link or links to make instruction more clear.

I believe it is sufficient.

File: docs/contributing/UPGRADE.md (hunk 19 above)
Comment (grossmannmartin) [reply docs/contributing/UPGRADE.md]: We renamed the file to `upgrade-monorepo.md`

File: docs/upgrade/UPGRADE-unreleased.md (hunk 20 above)
Comment (grossmannmartin) [reply docs/upgrade/UPGRADE-unreleased.md]: Indeed. I moved it and update instructions about writing upgrade.

File: docs/upgrade/UPGRADE-7.0.0-beta3.md (hunk 25 above)
Comment (grossmannmartin) [reply docs/upgrade/UPGRADE-7.0.0-beta3.md]: Indeed. I moved it and update instructions about writing upgrade.

File: docs/contributing/guidelines-for-writing-upgrade.md (hunk 22 above)
Comment (grossmannmartin) [reply docs/contributing/guidelines-for-writing-upgrade.md]: Note about the possibility to link specific changes added to upgrade guidelines.

File: docs/upgrade/UPGRADE-7.0.0-beta3.md (hunk 23 above)
Comment (grossmannmartin) [reply docs/upgrade/UPGRADE-7.0.0-beta3.md]: Moved to general instructions.

File: docs/upgrade/UPGRADE-7.0.0-beta3.md (hunk 24 above)
Comment (grossmannmartin) [reply docs/upgrade/UPGRADE-7.0.0-beta3.md]: Moved to general instructions.

File: docs/upgrade/UPGRADE-7.0.0-beta3.md (hunk 26 above)
Comment (grossmannmartin) [reply docs/upgrade/UPGRADE-7.0.0-beta3.md]: We rewrote this point to make it an instruction.

File: UPGRADE.md (hunk 27 above)
Comment (grossmannmartin) [reply UPGRADE.md]: Moved, thanks 👍

File: docs/contributing/guidelines-for-writing-upgrade.md [hunk 28]
```diff
@@ -17,3 +19,58 @@ Keep in mind that upgrade instructions are written for users that do not underst
...
+* shopsys/monorepo-tools
+* shopsys/plugin-interface
+* shopsys/product-feed-google
+* shopsys/product-feed-heureka
+* shopsys/product-feed-heureka-delivery
+* shopsys/product-feed-zbozi
+* shopsys/microservice-product-search
+* shopsys/microservice-product-search-export
+
+Each section must contain instructions relevant only to the package they cover and the section have to be ordered as they are in a list above.
```
Comment (vitek-rostislav) [inline docs/contributing/guidelines-for-writing-upgrade.md]:
```suggestion
Each section must contain instructions relevant only to the package they cover and the sections have to be ordered as they are in a list above.
```

**Comment by grossmannmartin** [comment]
Hi @vitek-rostislav, thank you for your review, I applied suggestion.
Branch is now rebased and ready to merge :)

File: docs/contributing/upgrading-monorepo.md [hunk 29]
```diff
@@ -0,0 +1,73 @@
...
+* `composer update`
+* `php phing clean`
+* `php phing db-migrations`
+* if you're experiencing some errors, you can always rebuild application and load demo data with `php phing build-demo-dev`
+
+## [From v7.0.0-beta4 to Unreleased](https://github.com/shopsys/shopsys/compare/v7.0.0-beta4...HEAD)
+- [#651 It's possible to add index prefix to elastic search](https://github.com/shopsys/shopsys/pull/651)
+    - either rebuild your Docker images with `docker-compose up -d --build` or add `ELASTIC_SEARCH_INDEX_PREFIX=''` to your `.env` files in the microservice root directories, otherwise all requests to the microservices will throw `EnvNotFoundException`
+- [#679 webserver container starts after php-fpm is started](https://github.com/shopsys/shopsys/pull/679)
+    - add `depends_on: php-fpm` into `webserver` service of your `docker-compose.yml` file so webserver will not fail on error `host not found in upstream php-fpm:9000`
```
Comment (PetrHeinz) [inline docs/contributing/upgrading-monorepo.md]: I've added `depends_on: php-fpm` to my service like this:

```yml
    webserver:
        image: nginx:1.13-alpine
        container_name: shopsys-framework-webserver
        depends_on: php-fpm
        volumes:
            - .:/var/www/html
            - ./docker/nginx/nginx.conf:/etc/nginx/conf.d/default.conf
        ports:
            - "8000:8080"
```

But after running `docker-compose up -d`, I've got this error message:

```
ERROR: The Compose file './docker-compose.yml' is invalid because:
services.webserver.depends_on contains an invalid type, it should be an array
```

The error message helped me to fix it, but I've felt like I've done what was suggested and it failed immediately. If I haven't run `docker-compose up` right after I've changed the file, I could have ended up wasting much more time on this.


```yml
    webserver:
        image: nginx:1.13-alpine
        container_name: shopsys-framework-webserver
        depends_on:
            - php-fpm
        volumes:
            - .:/var/www/html
            - ./docker/nginx/nginx.conf:/etc/nginx/conf.d/default.conf
        ports:
            - "8000:8080"
```

I suggest using the inline array syntax in the code snippet, so it's functional when copy-pasted:

```suggestion
    - add `depends_on: [php-fpm]` into `webserver` service of your `docker-compose.yml` file so webserver will not fail on error `host not found in upstream php-fpm:9000`
```

Comment (PetrHeinz) [reply docs/contributing/upgrading-monorepo.md]: Or it could be made consistent with the upgrade note for project implementations:

```suggestion
    - make sure your `webserver` service depends on `php-fpm` in your `docker-compose.yml` file so webserver will not fail on error `host not found in upstream php-fpm:9000` ([#679](https://github.com/shopsys/shopsys/pull/679))
```

Comment (vitek-rostislav) [reply docs/contributing/upgrading-monorepo.md]: thanks, I'll create a PR :slightly_smiling_face:
